src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/UpToDate/BuildUpToDateCheck.cs

```diff
@@ -56,32 +54,14 @@ internal sealed class BuildUpToDateCheck : OnceInitializedOnceDisposedUnderLockA
         private readonly ITelemetryService _telemetryService;
         private readonly IFileSystem _fileSystem;
 
+        private State _state = State.CreateEmpty();
+
         private IDisposable? _link;
-        private IComparable? _lastVersionSeen;
 
-        private bool _isDisabled = true;
         private bool _itemsChangedSinceLastCheck = true;
```

Comment:
This is one of two remaining fields that are mutated by `IsUpToDateAsync`. The fix for #4736 will remove this.

Reply:
With the lock removed, does this introduce a race for these?

Reply:
The `IsUpToDateAsync` check is still protected by a lock. Only the project subscription handler is unprotected now (though Dataflow prevents overlap).

There's as much of a race now as there was before for `_itemsChangedSinceLastCheck`.

This field is going away in the next few days anyway. Instead we'll have `LastItemChangedAtUtc` on the state snapshot.

That will leave just one mutable field, `LastCheckTimeUtc`, which is only read/written within the lock in `IsUpToDateAsync`.